src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Query/PropertyPages/QueryProjectPropertiesContext.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using Microsoft.VisualStudio.ProjectSystem.Query;
+using Microsoft.VisualStudio.ProjectSystem.Query.ProjectModel.Implementation;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Query
+{
+    /// <summary>
+    /// An implementation of <see cref="IProjectPropertiesContext"/> for use in the
+    /// Project Query API implementation.
+    /// </summary>
+    /// <remarks>
+    /// The point here is to capture the context for a part of a query execution in a
+    /// manner that can be passed from one provider to another and is also suitable as a
+    /// key into a cache (such as the <see cref="IPropertyPageQueryCache"/>).
+    /// </remarks>
+    internal sealed class QueryProjectPropertiesContext : IProjectPropertiesContext, IEquatable<QueryProjectPropertiesContext>
+    {
+        public static readonly QueryProjectPropertiesContext ProjectFile = new(isProjectFile: true, file: string.Empty, itemType: null, itemName: null);
```

Comment:
Note to self: add a doc comment.

This is meant to be a "well known" context representing the project file as a whole. Note that if an `IProjectPropertiesContext` has `IsProjectFile` property is set to `true` and the `ItemType` and `ItemName` properties are `null` then the properties system treats it as referring to the project file as a whole _regardless of the `File` property_. This lets us get away with setting it to the empty string and re-using the same instance across projects.